File: teddypet/src/main/java/fpt/teddypet/domain/enums/AttributeDisplayType.java
package fpt.teddypet.domain.enums;

import lombok.Getter;

@Getter
public enum AttributeDisplayType {
    TEXT("Ô nhập văn bản"),
    SELECT("Dropdown một lựa chọn"),
    CHECKBOX("Checkbox nhiều lựa chọn"),
    RADIO("Radio một lựa chọn"),
    MULTI_SELECT("Dropdown nhiều lựa chọn"),
    COLOR("Chọn màu sắc");

    private final String label;

    AttributeDisplayType(String label) {
        this.label = label;
    }

}

File: TeddyPet_FE/src/admin/types/enums/AttributeDisplayType.ts
export enum AttributeDisplayType {
  TEXT = 'TEXT',               // Ô nhập văn bản
  SELECT = 'SELECT',           // Dropdown một lựa chọn
  CHECKBOX = 'CHECKBOX',       // Checkbox nhiều lựa chọn
  RADIO = 'RADIO',             // Radio một lựa chọn
  MULTI_SELECT = 'MULTI_SELECT', // Dropdown nhiều lựa chọn
  COLOR = 'COLOR'              // Chọn màu sắc
}

// Display labels
export const AttributeDisplayTypeLabels: Record<AttributeDisplayType, string> = {
  [AttributeDisplayType.TEXT]: 'Ô nhập văn bản',
  [AttributeDisplayType.SELECT]: 'Dropdown một lựa chọn',
  [AttributeDisplayType.CHECKBOX]: 'Checkbox nhiều lựa chọn',
  [AttributeDisplayType.RADIO]: 'Radio một lựa chọn',
  [AttributeDisplayType.MULTI_SELECT]: 'Dropdown nhiều lựa chọn',
  [AttributeDisplayType.COLOR]: 'Chọn màu sắc',
};
